docs/ProfileDeveloperGuide.md

```diff
@@ -0,0 +1,430 @@
+# Table of Contents
+1. [Profiles](#Profiles)
+    1. [Fields](#Fields)
+    2. [Rules](#Rules)
+    3. [Persistence](#Persistence)
+    4. [Creation](#Creation)
+
+2. [Data types](#Data-Types)
+    1. [Numeric](#Numeric)
+    2. [Strings](#Strings)
+    3. [Temporal](#Temporal)
+
+3. [Predicate constraints](#Predicate-constraints)
+    1. [Theory](#Theory)
+    2. [General constraints](#General-constraints)
+        1. [equalTo](#predicate-equalto)
+        2. [inSet](#predicate-inset)
+        3. [null](#predicate-null)
+        4. [ofType](#predicate-oftype)
+    3. [Textual constraints](#Textual-constraints)
+        1. [matchingRegex](#predicate-matchingregex)
+        2. [containingRegex](#predicate-containingregex)
+        3. [ofLength](#predicate-oflength)
+        4. [longerThan](#predicate-longerthan)
+        5. [shorterThan](#predicate-shorterthan)
+        6. [aValid](#predicate-avalid)
+    4. [Numeric constraints](#Numeric-constraints)
+        1. [greaterThan](#predicate-greaterthan)
+        2. [greaterThanOrEqualTo](#predicate-greaterthanorequalto)
+        3. [lessThan](#predicate-lessthan)
+        4. [lessThanOrEqualTo](#predicate-lessthanorequalto)
+        5. [granularTo](#predicate-granularto)
+    5. [Temporal constraints](#Temporal-constraints)
+        1. [after](#predicate-after)
+        2. [afterOrAt](#predicate-afterorat)
+        3. [before](#predicate-before)
+        4. [beforeOrAt](#predicate-beforeorat)
+
+4. [Grammatical constraints](#Grammatical-Constraints)
+    1. [not](#not)
+    2. [anyOf](#anyOf)
+    3. [allOf](#allOf)
+    4. [if](#if)
+
+3. [Presentational constraints](#Presentational-Constraints)
+
+
+# Profiles
+
+**Data profiles** describe potential or real data. For instance, we could design a profile that describes a _user_account_ table, where:
```

Comment:
It's a good idea to have an example, but would be ideal if later it showed what this example profile would actually look like

Reply:
Agreed, I've added user-account into the example folder which contains this profile.